### PR TITLE
Adds the razorwire implant. Ready for review into merge

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Adds the razorwire arm implant. Classified as a weapon by beepsky

Restricted weapon, requires illegals 3 to print from rnd, in uplink for 15 TC
18 force sharp weapon with -100% armor penetration percentage (armor is twice as effective)

However, the whip has a 2 tile range. It can attack from 2 tiles away, instead of one.

Adds the range variable to all items. An item can be used by 1 tile per usual, but its range being increased should increase the range it can attack from. Only  the razorwire uses it, as the moment.

Negative armor penetration percentage attacks work now.

## Why It's Good For The Game

The introduction of potential 2 range weapons may be interesting. This weapon would be an interesting option for traitors, while not as strong as a gun, and weak vs armored officers, the ability to hit people from a tile away is not to be understated. However, will lose to a disabler or someone with armor. Can't be disarmed, however!
As for it being in RND, a potential later game upgrade for officers with illegals as a lethal weapon with a bit more range than a baton. Or for a traitor to mass produce, as an alternative to the emaged toolset.

### TODO:

- [x] Wait for #23876 to be merged.
- [x] Deconflict
- [x] Add to uplink
- [x] Add to rnd.

## Testing

Smacked stuff, confirmed inventory worked. Still want a round of TM or 2 to make sure clicks work right.

Smacked mobs from 2 tiles away with the weapon. And tables. And display cases, windows...

## Changelog
:cl:
add: Adds the Razorwire arm implant. A arm implant for 15 TC or in RND for illegals 3, that has a 2 tile range melee weapon with 18 force, and negative 100% armor penetration
/:cl:

### CODE REVIEW

File: code/_onclick/click.dm [hunk 1]
```diff
@@ -140,29 +138,83 @@
...
+			dummy.pass_flags |= PASSTABLE
+			dummy.invisibility = 120
+			for(var/i in 1 to reach) //Limit it to that many tries
+				var/turf/T = get_step(dummy, get_dir(dummy, there))
+				if(dummy.CanReach(there))
+					qdel(dummy)
+					return TRUE
+				if(!dummy.Move(T)) //we're blocked!
+					qdel(dummy)
+					return
```
Comment (DGamerL) [inline code/_onclick/click.dm:216]: `return FALSE`?

File: code/_onclick/click.dm [hunk 2]
```diff
@@ -140,29 +138,83 @@
...
+
+/atom/movable/proc/DirectAccess()
+	return list(src, loc)
+
+/mob/DirectAccess(atom/target)
+	return ..() + contents
+
+/mob/living/DirectAccess(atom/target)
+	return ..() + get_contents()
+
```
Comment (DGamerL) [inline code/_onclick/click.dm:195]:
```suggestion
```

File: code/_onclick/click.dm [hunk 3]
```diff
@@ -140,29 +138,83 @@
...
+ */
+/atom/movable/proc/CanReach(atom/ultimate_target, obj/item/tool, view_only = FALSE) //This might break mod storage. If it does, we hardcode mods / funny bag in here
+	var/list/direct_access = DirectAccess()
+	var/depth = 1 + (view_only ? STORAGE_VIEW_DEPTH : INVENTORY_DEPTH)
 
-	return
+	var/list/closed = list()
+	var/list/checking = list(ultimate_target)
+
+	while (checking.len && depth > 0)
```
Comment (DGamerL) [inline code/_onclick/click.dm:164]:
```suggestion
	while(length(checking) && depth > 0)
```

**Comment by Qwertytoforty** [comment]
Ready for TM

File: code/_onclick/click.dm [hunk 4]
```diff
@@ -140,29 +138,80 @@
...
 
 		return
+	else
+		if(!isturf(loc)) // This is going to stop you from telekinesing from inside a closet, but I don't shed many tears for that
+			return TRUE
 
-	if(!isturf(loc)) // This is going to stop you from telekinesing from inside a closet, but I don't shed many tears for that
-		return TRUE
+		if(W)
+			W.afterattack(A,src,0,params) // 0: not Adjacent
```
Comment (lewcc) [inline code/_onclick/click.dm:146]:
```suggestion
			W.afterattack(A, src, 0, params) // 0: not Adjacent
```

File: code/modules/surgery/organs/augments_arms.dm [hunk 5]
```diff
@@ -454,6 +470,72 @@
...
+
+/obj/item/melee/razorwire/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/two_handed, require_twohands = TRUE)
+
+/obj/item/melee/razorwire/examine_more(mob/user)
+	. = ..()
+	. += "<i>A byproduct of Cybersun Incorporated's mistakes turned concept, the Razorwire Spool is a remarkable accident in itself. \
+	It consists of a fine, thread-like laser capable of being manipulated and swung like a whip. Designed for ease of deployment, the wire originates from the wrist, \
+	allowing users with the implant to perform wide swings and precise cuts against soft targets. It's the same energy similar to other energy swords and daggers.<i>"
```
Comment (lewcc) [inline code/modules/surgery/organs/augments_arms.dm:501]:
```suggestion
	allowing users with the implant to perform wide swings and precise cuts against soft targets. It's the same energy found in other common energy weapons, such as swords and daggers.<i>"
```

Comment (Qwertytoforty) [reply code/modules/surgery/organs/augments_arms.dm]: done in all 3 places

File: code/modules/surgery/organs/augments_arms.dm [hunk 6]
```diff
@@ -454,6 +470,72 @@
...
+		even from a few steps away. However, results against anything more durable will heavily vary."
+	icon_state = "razorwire_weapon"
+	righthand_file = 'icons/mob/inhands/implants_righthand.dmi'
+	lefthand_file = 'icons/mob/inhands/implants_lefthand.dmi'
+	icon_state = "razorwire_weapon"
+	item_state = "razorwire"
+	w_class = WEIGHT_CLASS_BULKY
+	sharp = TRUE
+	force = 18
+	armour_penetration_percentage = -100 //This should mean armor is twice as effective. Should. testing needed
```
Comment (DGamerL) [inline code/modules/surgery/organs/augments_arms.dm:488]: Has the testing been done?

File: code/modules/surgery/organs/augments_arms.dm [hunk 7]
```diff
@@ -454,6 +470,72 @@
...
+	they ended up with a material that, while superheated and correctly composed, failed to maintain a solid blade shape. Curious about this error, \
+	Cybersun repeated the process, producing an energy as thin as a wire. After several prototypes, they achieved a long, energy-like thread. \
+	Further innovation allowed them to conceal this in a forearm-sized container, \
+	with a hand and wrist replacement made of the same durable material used to contain energy weapons. They would call it, the Razorwire.<i>"
+	. += "<i>Favored by assassins for their stealth and efficiency, Cybersun exercises discretion in its distribution, favoring clients in their good graces. \
+	It falls behind other energy weapons due to its thinner and more loose pressure, however it is praised more as a side-arm for unarmored soft targets.</i>"
+
+/obj/item/organ/internal/cyberimp/arm/razorwire/Extend(obj/item/item)
+	. = ..()
+
```
Comment (DGamerL) [inline code/modules/surgery/organs/augments_arms.dm:537]:
```suggestion
```

File: code/modules/surgery/organs/augments_arms.dm [hunk 8]
```diff
@@ -454,6 +470,72 @@
...
+	. += "<i>Cybersun's investment into energy weapon development inadvertently led to the Razorwire Spool. Initially attempting to create an Energy Sword, \
+	they ended up with a material that, while superheated and correctly composed, failed to maintain a solid blade shape. Curious about this error, \
+	Cybersun repeated the process, producing an energy as thin as a wire. After several prototypes, they achieved a long, energy-like thread. \
+	Further innovation allowed them to conceal this in a forearm-sized container, \
+	with a hand and wrist replacement made of the same durable material used to contain energy weapons. They would call it, the Razorwire.<i>"
+	. += "<i>Favored by assassins for their stealth and efficiency, Cybersun exercises discretion in its distribution, favoring clients in their good graces. \
+	It falls behind other energy weapons due to its thinner and more loose pressure, however it is praised more as a side-arm for unarmored soft targets.</i>"
+
+/obj/item/organ/internal/cyberimp/arm/razorwire/Extend(obj/item/item)
+	. = ..()
```
Comment (DGamerL) [inline code/modules/surgery/organs/augments_arms.dm:536]: Can't this be removed?

**Review by DGamerL — COMMENTED** [review]
Should be good for TM

File: code/modules/surgery/organs/autosurgeon.dm [hunk 9]
```diff
@@ -92,6 +92,23 @@
 	desc = "A single use autosurgeon that contains a pair of cybernetic meson eyes. A screwdriver can be used to remove it, but implants can't be placed back in."
 	uses = 1
 	starting_organ = /obj/item/organ/internal/eyes/cybernetic/meson
+/obj/item/autosurgeon/organ/syndicate/razorwire
```
Comment (Henri215) [inline code/modules/surgery/organs/autosurgeon.dm:95]:
```suggestion
	starting_organ = /obj/item/organ/internal/eyes/cybernetic/meson

/obj/item/autosurgeon/organ/syndicate/razorwire
```

File: code/modules/surgery/organs/augments_arms.dm (hunk 8 above)
Comment (Qwertytoforty) [reply code/modules/surgery/organs/augments_arms.dm:536]: Yeah, leftover before I moved some stuff to parent

File: code/modules/surgery/organs/autosurgeon.dm (hunk 9 above)
Comment (Qwertytoforty) [reply code/modules/surgery/organs/autosurgeon.dm:95]: done

File: code/modules/surgery/organs/augments_arms.dm (hunk 6 above)
Comment (DGamerL) [reply code/modules/surgery/organs/augments_arms.dm:488]: Nice, could you rewrite the comment a bit then?

Comment (Qwertytoforty) [reply code/modules/surgery/organs/augments_arms.dm:488]: oh, duh, one sec

**Comment by BAGELMENSK** [comment]
I am so hyped for this

File: code/_onclick/click.dm [hunk 10]
```diff
@@ -129,40 +129,98 @@
...
 		else
 			if(ismob(A))
 				changeNext_move(CLICK_CD_MELEE)
 			UnarmedAttack(A, 1)
-
 		return
 
-	if(!isturf(loc)) // This is going to stop you from telekinesing from inside a closet, but I don't shed many tears for that
-		return TRUE
+	else
```
Comment (S34NW) [inline code/_onclick/click.dm:141]: no need for this `else`, the above `if` always returns

File: code/_onclick/click.dm [hunk 11]
```diff
@@ -129,40 +129,98 @@
...
+	return ..() + get_contents()
+
+/proc/CheckToolReach(atom/movable/here, atom/movable/there, reach)
+	if(!here || !there)
+		return FALSE
+	switch(reach)
+		if(0)
+			return FALSE
+		if(1)
+			return FALSE //here.Adjacent(there)
```
Comment (S34NW) [inline code/_onclick/click.dm:210]:
```suggestion
		if(0, 1)
			return FALSE //here.Adjacent(there)
```

File: code/_onclick/cyborg.dm [hunk 12]
```diff
@@ -95,14 +95,12 @@
...
-			W.melee_attack_chain(src, A, params)
-			return
-		else
-			W.afterattack(A, src, 0, params)
-			return
-	return
+	if(can_reach(A,W))
+		W.melee_attack_chain(src, A, params)
+		return
+	else
```
Comment (S34NW) [inline code/_onclick/cyborg.dm:101]: no need for this else, above if returns always

File: code/_onclick/cyborg.dm [hunk 13]
```diff
@@ -95,14 +95,12 @@
...
 	// cyborgs are prohibited from using storage items so we can I think safely remove (A.loc && isturf(A.loc.loc))
-	if(isturf(A) || isturf(A.loc))
-		if(A.Adjacent(src)) // see adjacent.dm
-			W.melee_attack_chain(src, A, params)
-			return
-		else
-			W.afterattack(A, src, 0, params)
-			return
-	return
+	if(can_reach(A,W))
```
Comment (S34NW) [inline code/_onclick/cyborg.dm:98]:
```suggestion
	if(can_reach(A, W))
```

File: code/game/objects/items.dm [hunk 14]
```diff
@@ -132,6 +132,8 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
 	var/belt_icon = null
 	/// Holder var for the item outline filter, null when no outline filter on the item.
 	var/outline_filter
+	///In tiles, how far this weapon can reach; 1 for adjacent, which is default
```
Comment (S34NW) [inline code/game/objects/items.dm:135]:
```suggestion
	/// In tiles, how far this weapon can reach; 1 for adjacent, which is default
```

File: code/modules/surgery/organs/augments_arms.dm [hunk 15]
```diff
@@ -454,6 +455,63 @@
...
+	. += "<i>A byproduct of Cybersun Incorporated's mistakes turned concept, the Razorwire Spool is a remarkable accident in itself. \
+	It consists of a fine, thread-like laser capable of being manipulated and swung like a whip. Designed for ease of deployment, the wire originates from the wrist, \
+	allowing users with the implant to perform wide swings and precise cuts against soft targets. It's the same energy found in other common energy weapons, such as swords and daggers.<i>"
+	. += "<i>Cybersun's investment into energy weapon development inadvertently led to the Razorwire Spool. Initially attempting to create an Energy Sword, \
+	they ended up with a material that, while superheated and correctly composed, failed to maintain a solid blade shape. Curious about this error, \
+	Cybersun repeated the process, producing an energy as thin as a wire. After several prototypes, they achieved a long, energy-like thread. \
+	Further innovation allowed them to conceal this in a forearm-sized container, \
+	with a hand and wrist replacement made of the same durable material used to contain energy weapons. They would call it, the Razorwire.<i>"
```
Comment (S34NW) [inline code/modules/surgery/organs/augments_arms.dm:487]:
```suggestion
	with a hand and wrist replacement made of the same durable material used to contain energy weapons. They would call it, the Razorwire.</i>"
```

File: code/modules/surgery/organs/augments_arms.dm [hunk 16]
```diff
@@ -454,6 +455,63 @@
...
+	armour_penetration_percentage = -100 //This means that armor twice as effective against it
+	reach = 2
+	hitsound = 'sound/weapons/whip.ogg'
+	attack_verb = list("slashes", "whips", "lashes", "lacerates")
+
+/obj/item/melee/razorwire/examine_more(mob/user)
+	. = ..()
+	. += "<i>A byproduct of Cybersun Incorporated's mistakes turned concept, the Razorwire Spool is a remarkable accident in itself. \
+	It consists of a fine, thread-like laser capable of being manipulated and swung like a whip. Designed for ease of deployment, the wire originates from the wrist, \
+	allowing users with the implant to perform wide swings and precise cuts against soft targets. It's the same energy found in other common energy weapons, such as swords and daggers.<i>"
```
Comment (S34NW) [inline code/modules/surgery/organs/augments_arms.dm:482]:
```suggestion
	allowing users with the implant to perform wide swings and precise cuts against soft targets. It's the same energy found in other common energy weapons, such as swords and daggers.</i>"
```

File: code/modules/surgery/organs/augments_arms.dm [hunk 17]
```diff
@@ -454,6 +455,63 @@
...
+	action_icon = list(/datum/action/item_action/organ_action/toggle = 'icons/obj/surgery.dmi')
+	action_icon_state = list(/datum/action/item_action/organ_action/toggle = "razorwire")
+	origin_tech = "combat=5;biotech=5;syndicate=2"
+	stealth_level = 1 // Hidden from health analyzers
+
+/obj/item/organ/internal/cyberimp/arm/razorwire/examine_more(mob/user)
+	. = ..()
+	. += "<i>A byproduct of Cybersun Incorporated's mistakes turned concept, the Razorwire Spool is a remarkable accident in itself. \
+	It consists of a fine, thread-like laser capable of being manipulated and swung like a whip. Designed for ease of deployment, the wire originates from the wrist, \
+	allowing users with the implant to perform wide swings and precise cuts against soft targets. It's the same energy found in other common energy weapons, such as swords and daggers.<i>"
```
Comment (S34NW) [inline code/modules/surgery/organs/augments_arms.dm:506]:
```suggestion
	allowing users with the implant to perform wide swings and precise cuts against soft targets. It's the same energy found in other common energy weapons, such as swords and daggers.</i>"
```

File: code/modules/surgery/organs/augments_arms.dm [hunk 18]
```diff
@@ -454,6 +455,63 @@
...
+	. += "<i>Cybersun's investment into energy weapon development inadvertently led to the Razorwire Spool. Initially attempting to create an Energy Sword, \
+	they ended up with a material that, while superheated and correctly composed, failed to maintain a solid blade shape. Curious about this error, \
+	Cybersun repeated the process, producing an energy as thin as a wire. After several prototypes, they achieved a long, energy-like thread. \
+	Further innovation allowed them to conceal this in a forearm-sized container, \
+	with a hand and wrist replacement made of the same durable material used to contain energy weapons. They would call it, the Razorwire.<i>"
```
Comment (S34NW) [inline code/modules/surgery/organs/augments_arms.dm:511]:
```suggestion
	with a hand and wrist replacement made of the same durable material used to contain energy weapons. They would call it, the Razorwire.</i>"
```

File: code/modules/surgery/organs/augments_arms.dm [hunk 19]
```diff
@@ -454,6 +455,63 @@
...
+	desc = "An integrated spool of razorwire, capable of being used as a weapon when whipped at your foes. \
+		Built into the back of your hand, try your best to not get it tangled."
+	contents = newlist(/obj/item/melee/razorwire)
+	icon_state = "razorwire"
+	action_icon = list(/datum/action/item_action/organ_action/toggle = 'icons/obj/surgery.dmi')
+	action_icon_state = list(/datum/action/item_action/organ_action/toggle = "razorwire")
+	origin_tech = "combat=5;biotech=5;syndicate=2"
+	stealth_level = 1 // Hidden from health analyzers
+
+/obj/item/organ/internal/cyberimp/arm/razorwire/examine_more(mob/user)
```
Comment (S34NW) [inline code/modules/surgery/organs/augments_arms.dm:502]: To save copy-paste, can you perhaps reference the item (through contents) and use the result of calling `examine_more()` on it?

File: code/modules/surgery/organs/autosurgeon.dm [hunk 20]
```diff
@@ -93,6 +93,24 @@
 	uses = 1
 	starting_organ = /obj/item/organ/internal/eyes/cybernetic/meson
 
+/obj/item/autosurgeon/organ/syndicate/razorwire
+	desc = "A single use autosurgeon that contains a Razorwire arm implant. A screwdriver can be used to remove it, but implants can't be placed back in."
+	uses = 1
+	starting_organ = /obj/item/organ/internal/cyberimp/arm/razorwire
+
+/obj/item/autosurgeon/organ/syndicate/razorwire/examine_more(mob/user)
```
Comment (S34NW) [inline code/modules/surgery/organs/autosurgeon.dm:101]: same here with reference to `starting_organ`?

File: code/modules/surgery/organs/autosurgeon.dm [hunk 21]
```diff
@@ -93,6 +93,24 @@
...
+	. = ..()
+	. += "<i>A byproduct of Cybersun Incorporated's mistakes turned concept, the Razorwire Spool is a remarkable accident in itself. \
+	It consists of a fine, thread-like laser capable of being manipulated and swung like a whip. Designed for ease of deployment, the wire originates from the wrist, \
+	allowing users with the implant to perform wide swings and precise cuts against soft targets. It's the same energy found in other common energy weapons, such as swords and daggers.<i>"
+	. += "<i>Cybersun's investment into energy weapon development inadvertently led to the Razorwire Spool. Initially attempting to create an Energy Sword, \
+	they ended up with a material that, while superheated and correctly composed, failed to maintain a solid blade shape. Curious about this error, \
+	Cybersun repeated the process, producing an energy as thin as a wire. After several prototypes, they achieved a long, energy-like thread. \
+	Further innovation allowed them to conceal this in a forearm-sized container, \
+	with a hand and wrist replacement made of the same durable material used to contain energy weapons. They would call it, the Razorwire.<i>"
```
Comment (S34NW) [inline code/modules/surgery/organs/autosurgeon.dm:110]:
```suggestion
	with a hand and wrist replacement made of the same durable material used to contain energy weapons. They would call it, the Razorwire.</i>"
```

File: code/modules/surgery/organs/autosurgeon.dm [hunk 22]
```diff
@@ -93,6 +93,24 @@
...
+/obj/item/autosurgeon/organ/syndicate/razorwire
+	desc = "A single use autosurgeon that contains a Razorwire arm implant. A screwdriver can be used to remove it, but implants can't be placed back in."
+	uses = 1
+	starting_organ = /obj/item/organ/internal/cyberimp/arm/razorwire
+
+/obj/item/autosurgeon/organ/syndicate/razorwire/examine_more(mob/user)
+	. = ..()
+	. += "<i>A byproduct of Cybersun Incorporated's mistakes turned concept, the Razorwire Spool is a remarkable accident in itself. \
+	It consists of a fine, thread-like laser capable of being manipulated and swung like a whip. Designed for ease of deployment, the wire originates from the wrist, \
+	allowing users with the implant to perform wide swings and precise cuts against soft targets. It's the same energy found in other common energy weapons, such as swords and daggers.<i>"
```
Comment (S34NW) [inline code/modules/surgery/organs/autosurgeon.dm:105]:
```suggestion
	allowing users with the implant to perform wide swings and precise cuts against soft targets. It's the same energy found in other common energy weapons, such as swords and daggers.</i>"
```

**Review by S34NW — CHANGES_REQUESTED** [review]

File: code/_onclick/click.dm (hunk 10 above)
Comment (Qwertytoforty) [reply code/_onclick/click.dm:141]: done

File: code/_onclick/cyborg.dm (hunk 12 above)
Comment (Qwertytoforty) [reply code/_onclick/cyborg.dm:101]: done

File: code/modules/surgery/organs/augments_arms.dm (hunk 19 above)
Comment (Qwertytoforty) [reply code/modules/surgery/organs/augments_arms.dm:502]: done!

File: code/modules/surgery/organs/autosurgeon.dm (hunk 20 above)
Comment (Qwertytoforty) [reply code/modules/surgery/organs/autosurgeon.dm:101]: done!

**Review by S34NW — APPROVED** [review]
This has been stable for a while, it's an overall minor change so is good to go

**Pull request merged**